spark3/src/main/java/org/apache/iceberg/spark/SparkSessionCatalog.java

```diff
@@ -242,6 +242,9 @@ public void renameTable(Identifier from, Identifier to) throws NoSuchTableExcept
 
   @Override
   public final void initialize(String name, CaseInsensitiveStringMap options) {
+    if (options.containsKey("type") && options.get("type").equalsIgnoreCase("hive") && options.containsKey("uri")) {
+      throw new UnsupportedOperationException("Passing hive catalog uri through spark.sql.catalog.<catalog_name>.uri is not allowed̉.");
```

Comment:
I would change this to be a bit more specific, "Cannot set an alternative uri when using the SparkSessionCatalog, make an alternative SparkCatalog if the Spark Session catalog and Iceberg catalog should contact different metastores."

We also add in a test to check that this works as expected :)

Reply:
Sure, have added a test for this scenario and changes the exception message. Please check.